gems/aws-sdk-s3/lib/aws-sdk-s3/presigner.rb

```diff
@@ -231,17 +231,23 @@ def sign_but_dont_send(
           end
           http_req.endpoint.query = query.join('&') unless query.empty?
 
+          signing_algorithm = :sigv4
+
           # If it's an ARN, get the resolved region and service
           if (arn = context.metadata[:s3_arn])
             region = arn[:resolved_region]
             service = arn[:arn].service
+            region = arn[:arn].is_a?(MultiRegionAccessPointARN) ? '*': arn[:resolved_region]
+            signing_algorithm = arn[:arn].is_a?(MultiRegionAccessPointARN) ? :sigv4a : :sigv4
```

Comment:
Sigv4 is already the default here

Reply:
Wanted this line to match the above - I think its easier to read here than:
```
 signing_algorithm = : sigv4a if arn[:arn].is_a?(MultiRegionAccessPointARN) 
```